### PR TITLE
Feature/configurator async

### PR DESCRIPTION
**Pull request by JeGoi**
# Description
Add async restart for multiple service

# Impacts
Make venom configurator stable for 12.x
Restart other services with async

# Delete branch after merge
YES

# Checklist
- [ ] Add multiple tests
- [ ] Add pipelines tests

### CODE REVIEW

File: t/venom/lib/pf_api_service_start_async.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+executor: pf_api_service_start_async
+input:
+  service: ""
+  time_to_sleep: 0
+  i_retry: 40
+  i_delay: 20
```
Comment (nqb) [inline t/venom/lib/pf_api_service_start_async.yml]: The retry/delay is different here, is it on purpose?

File: t/venom/lib/pf_api_service_start_async.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+steps:
+- type: pf_api_service_action
+  service: "{{.input.service}}"
+  action: "restart"
```
Comment (nqb) [inline t/venom/lib/pf_api_service_start_async.yml]: this action 'restart' doesn't match name of executor 'start"

File: t/venom/test_suites/common/restart_haproxy_admin.yml [hunk 3]
```diff
@@ -1,11 +1,6 @@
-name: Restart haproxy-admin service
+name: Restart haproxy-portal service
```
Comment (nqb) [inline t/venom/test_suites/common/restart_haproxy_admin.yml]: "admin" has been replaced by "portal". I don't think it's wanted.

File: t/venom/test_suites/common/restart_haproxy_admin.yml [hunk 4]
```diff
@@ -1,11 +1,6 @@
...
-- name: restart_haproxy_admin_service
+- name: restart_haproxy_portal_service
   steps:
-  - type: exec
-    script: systemctl restart packetfence-haproxy-admin
-
-    # let service restarts
-  - type: exec
-    script: sleep 5
+  - type: pf_api_service_start_async
```
Comment (nqb) [inline t/venom/test_suites/common/restart_haproxy_admin.yml]: I expect to see `pf_api_system_service_restart_async` if we are doing a restart

File: t/venom/test_suites/common/restart_haproxy_portal.yml [hunk 5]
```diff
@@ -1,17 +1,6 @@
...
-  - type: http
-    method: POST
-    url: '{{.pfserver_webadmin_url}}/api/v1/service/haproxy-portal/restart'
-    ignore_verify_ssl: true
-    headers:
-      "Authorization": "{{.get_login_token.result.token}}"
-      "Content-Type": "application/json"
-    assertions:
-      - result.statuscode ShouldEqual 200
+  - type: pf_api_service_start_async
```
Comment (nqb) [inline t/venom/test_suites/common/restart_haproxy_portal.yml]: I expect to see `pf_api_system_service_restart_async` if we are doing a restart

File: t/venom/test_suites/configurator/50_run_configurator_step4.yml [hunk 6]
```diff
@@ -36,55 +23,22 @@ testcases:
       - result.statuscode ShouldEqual 200
       - result.bodyjson.message ShouldEqual "Updated systemd for pf"
 
-# service restarted by hand because using API cause Venom failure (EOF)
 - name: restart_haproxy_admin_service
   steps:
-  - type: exec
-    script: systemctl restart packetfence-haproxy-admin
+  - type: pf_api_service_start_async
```
Comment (nqb) [inline t/venom/test_suites/configurator/50_run_configurator_step4.yml]: I expect to see `pf_api_system_service_restart_async` if we are doing a restart

File: t/venom/test_suites/common/restart_haproxy_admin.yml [hunk 7]
```diff
@@ -1,11 +1,6 @@
...
+    service: 'haproxy-portal'
```
Comment (nqb) [inline t/venom/test_suites/common/restart_haproxy_admin.yml]: haproxy-admin

File: t/venom/test_suites/configurator_inline_l2/50_run_configurator_step4.yml [hunk 8]
```diff
@@ -24,55 +23,22 @@ testcases:
       - result.statuscode ShouldEqual 200
       - result.bodyjson.message ShouldEqual "Updated systemd for pf"
 
-# service restarted by hand because using API cause Venom failure (EOF)
 - name: restart_haproxy_admin_service
   steps:
-  - type: exec
-    script: systemctl restart packetfence-haproxy-admin
+  - type: pf_api_service_start_async
```
Comment (nqb) [inline t/venom/test_suites/configurator_inline_l2/50_run_configurator_step4.yml]: same remark as for previous configurator test suite

File: t/venom/test_suites/wired_dot1x_eap_tls_pki/41_restart_haproxy-admin_service.yml [hunk 9]
```diff
@@ -1,7 +1,6 @@
...
+name: Restart haproxy-portal service
 testcases:
-# service restarted by hand because using API cause Venom failure (EOF)
-- name: restart_packetfence_haproxy-admin_service
+- name: restart_haproxy_portal_service
   steps:
-  - type: systemctl_service_restart
-    service: packetfence-haproxy-admin
+  - type: pf_api_service_start_async
+    service: 'haproxy-portal'
```
Comment (nqb) [inline t/venom/test_suites/wired_dot1x_eap_tls_pki/41_restart_haproxy-admin_service.yml]: same remark as for previous configurator test suite

File: t/venom/test_suites/wired_dot1x_eap_tls_pki/42_restart_haproxy-portal_service.yml [hunk 10]
```diff
@@ -1,6 +1,6 @@
 name: Restart haproxy-portal service
 testcases:
-- name: restart_haproxy-portal_service
+- name: restart_haproxy_portal_service
   steps:
-  - type: pf_api_service_restart
-    service: haproxy-portal
+  - type: pf_api_service_start_async
```
Comment (nqb) [inline t/venom/test_suites/wired_dot1x_eap_tls_pki/42_restart_haproxy-portal_service.yml]: same remark as for previous configurator test suite

File: t/venom/lib/pf_api_service_start_async.yml (hunk 1 above)
Comment (JeGoi) [reply t/venom/lib/pf_api_service_start_async.yml]: What do you mean? The name? if yes, it is to make a difference between to the executor.

Comment (nqb) [reply t/venom/lib/pf_api_service_start_async.yml]: The retry/delay value is different from all other executors you create in this PR.

Comment (JeGoi) [reply t/venom/lib/pf_api_service_start_async.yml]: I do not remember. Let's make it as the other ones. Thanks

File: t/venom/test_suites/common/restart_haproxy_admin.yml (hunk 3 above)
Comment (nqb) [inline t/venom/test_suites/common/restart_haproxy_admin.yml]:
```suggestion
name: Restart haproxy-admin service
```

File: t/venom/test_suites/common/restart_haproxy_admin.yml [hunk 11]
```diff
@@ -1,11 +1,6 @@
-name: Restart haproxy-admin service
+name: Restart haproxy-portal service
 testcases:
-# service restarted by hand because using API cause Venom failure (EOF)
-- name: restart_haproxy_admin_service
+- name: restart_haproxy_portal_service
```
Comment (nqb) [inline t/venom/test_suites/common/restart_haproxy_admin.yml]:
```suggestion
- name: restart_haproxy_admin_service
```

**Review by nqb — CHANGES_REQUESTED** [review]

**Review by nqb — APPROVED** [review]

**Pull request merged**